gpt4all-chat/modellist.cpp

```diff
@@ -952,11 +960,21 @@ void ModelList::updateData(const QString &id, const QVector<QPair<int, QVariant>
         }
 
         // Extra guarantee that these always remains in sync with filesystem
-        const QFileInfo fileInfo(info->dirpath + info->filename());
+        QString modelPath = info->dirpath + info->filename();
+        const QFileInfo fileInfo(modelPath);
         info->installed = fileInfo.exists();
         const QFileInfo incompleteInfo(incompleteDownloadPath(info->filename()));
         info->isIncomplete = incompleteInfo.exists();
 
+        // check installed, discovered/sideloaded models only (including clones)
+        if (!info->checkedEmbeddingModel && !info->isEmbeddingModel && info->installed
+            && (info->isDiscovered() || info->description().isEmpty()))
+        {
+            // read GGUF and decide based on model architecture
+            info->isEmbeddingModel = LLModel::Implementation::isEmbeddingModel(modelPath.toStdString());
```

Comment:
This is a little scary since we're doing this inside of a locked mutex that is accessed by the main thread. So we're potentially tying up the main thread if this loading of the model takes a long time it will result in the GUI being unresponsive.